src/core/utils.ts

```diff
@@ -74,15 +76,27 @@ export interface Story2FilenameParams {
   viewport: Viewport | null;
   namespace: string | null;
   knobs: StoredKnobs | null;
+  filePattern?: string | null;
 }
 
 export const story2filename = (params: Story2FilenameParams) => {
   const ns = params.namespace ? `_${params.namespace}` : '';
   const vp = params.viewport ? `-${viewport2string(params.viewport)}` : '';
   const knobs = params.knobs ? `-${knobs2string(params.knobs)}` : '';
-  const filename = `${filenamify(`${params.kind}-${params.story}${knobs}${ns}${vp}`)}`;
 
-  return `${filename}.png`;
+  const replacements = {
+    kind: params.kind,
+    story: params.story,
+    knobs,
+    ns,
+    vp
+  };
+  
+  const filename =
+    (params.filePattern || DEFAULT_FILE_PATTERN)
+     .replace(/\{(\w+)\}/g, (_,key) => replacements[key]);
```

Comment:
I would like a unit test for this change if possible :pray:

Reply:
One already exists - for default pattern. More to come.

Reply:
Done! And please double check it.